src/Uploadcare/Api.php

```diff
@@ -194,6 +197,12 @@ public function __construct(
         $this->cdn_secret_token = $cdn_secret_token;
         $this->cdn_provider = $cdn_provider;
 
+        if ($this->cdn_secret_token) {
+            if(empty($this->cdn_provider)) {
+                throw new \Exception('CDN provider must not be empty');
+            }
```

Comment:
this is not needed here, as we have a default setting and proper exception down the road.